src/App.tsx

```diff
@@ -8,32 +10,45 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
+  const [clockName, setClockName] = useState('Clock-0');
+  const [hasClock, setHasClock] = useState(true);
+
+  const onRightClick = (event: MouseEvent) => {
+    event.preventDefault();
+    setHasClock(false);
+  };
+
+  const onClick = () => setHasClock(true);
+
+  useEffect(() => {
+    if (hasClock && clockName !== 'Clock-0') {
```

Comment:
`Clock-0` looks hardcoded here. What if we will change the default name later? Better to create variable for that